addon/mixins/shared-stylist.js

```diff
@@ -1,79 +1,79 @@
+  _styleInit: Ember.on('willInsertElement', function() {
+    this._setStyle(); 
   }),
-  _style: on('init',observer(..._styleProperties, function() {
-    this._setStyle();
-  })),
-  _setStyle() {
-    const styleProperties = this.getProperties(..._styleProperties);
-    /**
-     * Provides a per-type styler that allows for some sensible defaults
-     * @param  {string} style The style property being evaluated
-     * @param  {string} value The suggested value for this style property
-     * @return {string}       A mildly processed/improved variant on the input
-     */
-    const stylist = (style, value) => {
-      switch(style) {
-        case 'fontSize':
-        case 'width':
-        case 'minWidth':
-        case 'maxWidth':
-          return sizer(value);
-        case 'height':
-          let width = this.get('width');
-          if(!width || String(width).substr(-2) !== 'px') {
-            return sizer(value);
-          }
-          width = width.substr(0,width.length - 2);
-          if(value === 'golden') {
-             return width / GOLDEN_RATIO + 'px';
-          } else if (value === 'square' && this.get('width')) {
-            return width / ASPECT_RATIO + 'px';
-          } else {
-            return sizer(value);
-          }
-          return value;
-        default:
-          if(new A(['undefined','null']).contains(typeOf(value))) {
-            return null;
-          }
-          return value;
-      }
-    };
-    run.schedule('afterRender', this, function(){
-      let style = this.get('element').style;
-      let whitelist = new A(this.get('_styleWhitelist'));
-      keys(styleProperties).map(item => {
-        if(whitelist.contains(item)) {
-          style[item] = securitize(stylist(item,styleProperties[item]));
+
+  _styleObserver: Ember.observer('style', function() {
+    this._setStyle(); 
+  }),
+
+  _setStyle: function() {
+    var style = this.get('style');
+
+    var domStyle = this.get('element').style;
+
+    var newKeys = [];
+    var removedKeys = this._lastKeys;
+
+    for (var key in style) {
+      if (style.hasOwnProperty(key)) {
+        var value = style[key];
+
+        // clean up the key and value
+        key = Ember.String.dasherize(key);
+        value = this._escape(value);
+
+        // set the dom style
+        domStyle.setProperty(key, value);
+
+        // deal with keys that have been removed since last time
+        newKeys.push(key);
+        var index = removedKeys.indexOf(key);
+        if (index >= 0)
+        {
+          removedKeys.splice( index, 1 );
         }
-      });
-    });
+      }
+    }
+    // remove old styles that we don't need any more
+    for (var removedIndex in removedKeys) {
+      domStyle.removeProperty(removedKeys[removedIndex]);
+    }
+    this._lastKeys = newKeys;
   },
-});
 
-SharedStylist[Ember.NAME_KEY] = 'Shared Stylist';
-export default SharedStylist;
+  _escape: function(data) {
+    if (data instanceof Ember.Handlebars.SafeString)
+    {
```

Comment:
I thought about doing this but if I'm not mistaken, using this "under the radar" approach to style also means that Ember will not process the SafeString before putting it in the DOM and therefore someone going to the now unnecessary step of putting it into a SafeString not only is inefficient but actually breaks. The only thing SafeString was doing was allowing the string to be escaped at some later time. That's why you'll see that all I'm doing is getting it back to a string so that -- like other string types -- it can just be escaped (which the DOM will recognise; unlike SafeString)